File: PetHealth/app/src/main/java/pe/edu/upc/pethealth/network/PetHealthServices.java
package pe.edu.upc.pethealth.network;

import com.google.gson.JsonObject;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.HTTP;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface PetHealthServices {
    @POST(EndpointUrls.LOGIN)
    Call<RestView<JsonObject>> login(@Body JsonObject requestBody);

    @HTTP(method = "GET", path = EndpointUrls.APPOINTMENTS, hasBody = true)
    Call<RestView<JsonObject>> getAppts(@Body JsonObject body, @Path("user_id") int userId);
}

File: PetHealth/clinic-app/src/main/java/pe/edu/upc/phclinic/network/PetHealthServices.kt
package pe.edu.upc.phclinic.network

import com.google.gson.JsonObject
import pe.edu.upc.lib.models.ApptModel
import pe.edu.upc.phclinic.network.RestClient.Companion.APPOINTMENTS
import pe.edu.upc.phclinic.network.RestClient.Companion.LOGIN
import pe.edu.upc.phclinic.network.RestClient.Companion.SIGNUP_USER
import retrofit2.Call
import retrofit2.http.*

interface PetHealthServices {
    @POST(LOGIN)
    fun login(@Body requestBody: JsonObject): Call<RestView<JsonObject>>

    @GET(APPOINTMENTS)
    fun getAppts(@Header("access_token") accessToken: String, @Path("user_id") userId: Int): Call<ApptModel.Response>

    @POST(SIGNUP_USER)
    fun signup(@Body requestBody: JsonObject): Call<RestView<JsonObject>>
}